core-bundle/src/Search/Indexer/DefaultIndexer.php

```diff
@@ -93,6 +101,17 @@ public function index(Document $document): void
 
         $this->framework->initialize();
 
+        // Try to extract page id
+        try {
+            $parameters = $this->urlMatcher->match($document->getUri()->getPath());
+
+            if (\array_key_exists('pageModel', $parameters) && $parameters['pageModel'] instanceof PageModel) {
+                $pageId = (int) $parameters['pageModel']->id;
+            }
+        } catch (ExceptionInterface $exception) {
+            $pageId = 0;
```

Comment:
What if there is no exception in the try block and the if condition does not match? Then `$pageId` will be undefined, won't it?